include/boost/hana/sum.hpp

```diff
@@ -23,28 +23,28 @@ Distributed under the Boost Software License, Version 1.0.
 
 
 BOOST_HANA_NAMESPACE_BEGIN
+    //! @cond
     template <typename M>
-    struct sum_t {
+    template <typename Xs>
+    constexpr decltype(auto) sum_t<M>::operator()(Xs&& xs) const {
     #ifndef BOOST_HANA_CONFIG_DISABLE_CONCEPT_CHECKS
         static_assert(hana::Monoid<M>::value,
         "hana::sum<M> requires 'M' to be a Monoid");
     #endif
 
-        template <typename Xs>
-        constexpr decltype(auto) operator()(Xs&& xs) const {
-            using S = typename hana::tag_of<Xs>::type;
-            using Sum = BOOST_HANA_DISPATCH_IF(sum_impl<S>,
-                hana::Foldable<S>::value
-            );
+        using S = typename hana::tag_of<Xs>::type;
+        using Sum = BOOST_HANA_DISPATCH_IF(sum_impl<S>,
+            hana::Foldable<S>::value
+        );
 
-        #ifndef BOOST_HANA_CONFIG_DISABLE_CONCEPT_CHECKS
```

Comment:
Could you please group the two concept checks inside the same `#ifndef` please? Here and elsewhere.

Reply:
Ping. I'm good to merge this once this is done. I'd like to avoid having multiple `#ifdefs` for `BOOST_HANA_CONFIG_DISABLE_CONCEPT_CHECKS` in the same function when that makes sense.

Reply:
I had pushed for this, but apparently I missed the second overloads for `monadic_fold_left/right`.
I rechecked them all and pushed again. Should be fixed now. :sweat_smile: